conans/model/scm.py

```diff
@@ -27,6 +34,7 @@ def __init__(self, conanfile):
             self.password = data.get("password")
             self.subfolder = data.get("subfolder")
             self.submodule = data.get("submodule")
+            self._shallow = SCMData._boolean(data.get("shallow"), None)
```

Comment:
Should implement the same for `verify_ssl`, it is also a boolean value

Reply:
Will create PR asap this one gets reviewed and merged

Reply:
Cant we assume that they are always python booleans? The verify_ssl sounds like a bug then? Maybe should be targeted to 1.18.2 too?

Reply:
Yes, it is a bug. Asap as this PR gets reviewed I will create the PR for the `verify_ssl` targeting `1.18.2` with the same implementation as this one.

Reply:
Btw, we can't assume it would be actual Python booleans as long as we use `json.dumps` to convert it to string. It is not posible to output `True/False` (without quotes) so we must be able to read `True` (probably user input) and `"True"` (the value be get after exporting the recipe)...

We can change the way we are serializing to string to print `True/False` booleans, but it should be something for `1.19`, not for this minor version I think.

Reply:
I think the better is:

- Check the types in the __init__
- Don't use json.dumps to serialize, just build it manually taking into account the boolean fields, kind of:
       `'{%s}' % ', '.join(['"%s": "%s"' % (k, v) for k, v in a.items()])`
  I don't trust much the `str(data)` and the quotes that way are simple ones.
- As I said in the other comment, if the boolean value is the default, do not serialize it.
- Doing the above we avoid checking "True", true, "None", none and so on